### PR TITLE
[minor] add missing slt tests for count(partitioned,aggregated, aggregated cube)

### PR DESCRIPTION
**Pull request by buraksenn**
## Which issue does this PR close?
-

## Rationale for this change
I saw TODO's in the codebase that these did not have any slt tests

## What changes are included in this PR?
Several slt test cases for count(partitioned,aggregated, aggregated cube)

## Are these changes tested?
They are only slt tests

## Are there any user-facing changes?
No

### CODE REVIEW

**Review by alamb — COMMENTED** [review]
Thank you @buraksenn  🙏 

You mention a TODO in the PR description. Should we also remove the TPDP?

**Review by alamb — APPROVED** [review]

**Comment by buraksenn** [comment]
> Thank you @buraksenn 🙏
> 
> You mention a TODO in the PR description. Should we also remove the TPDP?

Thanks @alamb. I've already deleted them they are in the slt file actually

File: datafusion/sqllogictest/test_files/aggregate.slt [hunk 1]
```diff
@@ -2179,11 +2179,62 @@ SELECT COUNT(c1), COUNT(c2) FROM test
 ----
 4 4
 
-# TODO: count_partitioned
```
Comment (alamb) [inline datafusion/sqllogictest/test_files/aggregate.slt:2182]: Got it -- it is this 👍

**Comment by berkaysynnada** [comment]
LGTM, double-checked in postgre

**Pull request merged**